clip/README.md

```diff
@@ -0,0 +1,104 @@
+# (OpenAI) CLIP
+
+An example of visual-language representation learning using MLX.
+
+CLIP, for Contrastive Language-Image Pre-training, is a powerful representation learning method that aims to learn shared embedding space for both images and text.
+
+Paper: https://arxiv.org/pdf/2103.00020.pdf
+
+### Setup
+
+Install the dependencies:
+
+```
+pip install -r requirements.txt
+```
+
+Next, download the model configuration and weights. To download the model, it is necessary to specify the desired HuggingFace configuration and checkpoint. In this example, we will use ```
+openai/clip-vit-base-patch32```.
+
+```
+python download.py openai/clip-vit-base-patch32
+```
+If the download is successful, the folder ```weights``` will appear. In the ```weights``` folder, there will be the HuggingFace configuration and weights for ```openai/clip-vit-base-patch32```.
+
+Now, convert the PyTorch weights to MLX with:
+
+```
+python convert.py --torch-path weights/openai/clip-vit-base-patch32 --mlx-path weights/mlx/openai/clip-vit-base-patch32
+```
+
+By default, the conversion script will make the directory `mlx` folder within `weights`.
+In `weights/mlx/openai/clip-vit-base-patch32`, the script will save
+the converted `weights.npz`,  `config.json`,  `vocab.json`, and `merges.txt`.
+
+### Run
+
+Once you've converted the weights to MLX format, you can use the
+CLIP model to embed images and text. 
+
+```python
+from pathlib import Path
+from model import CLIPModel
+from PIL import Image
+from tokenizer import CLIPTokenizer
+from image_processor import CLIPImageProcessor
+
+MODEL: str = "openai/clip-vit-base-patch32"
+CONVERTED_CKPT_PATH: str = f"weights/mlx/{MODEL}"
+
+# Load pretrained MLX CLIPModel
+mlx_clip = CLIPModel.from_pretrained(Path(CONVERTED_CKPT_PATH))
+# Load input tokenizer and transformers image (pre)processor
+tokenizer = CLIPTokenizer.from_pretrained(Path(CONVERTED_CKPT_PATH))
+img_processor = CLIPImageProcessor.from_pretrained(Path(CONVERTED_CKPT_PATH))
```

Comment:
We shoudl simplify this to a single call to a `load` like this:

```python
model, tokenizer, image_processor = load(string_path)
```

And use a `string_path` to load the model not a `Path`. Simplifies a lot for the caller.

Reply:
I will implement something similar to the suggestion :)